### PR TITLE
Run generate-api-types in precommit hook when models have changed; Run CI on push for all branches including main

### LINKED ISSUE #1340: Run CI on every push, not just PR
Also, run `yarn generate-api-types` after linting and verify there aren't any uncommitted changes (e.g. with `git status --porcelain`).

This will stop us from having merge conflicts if commit-hooks are skipped for whatever reason.

### PR DESCRIPTION
**Pull request by bttf**
Closes GB-203
Closes #1340

### CODE REVIEW

**Review by tinahollygb — CHANGES_REQUESTED** [review]
I don't think this is doing what #1340 says. My understanding of the ticket criteria is we should do the following **on CI**:

- run `yarn generate-api-types`
- run `git status --porcelain` 
- fail CI if it is not clean

The changes you've added will only add another task to the pre-commit hooks. This doesn't actually solve our problem because the current problem is not everybody is actually running the pre-commit hooks and the rest of us only find out after we've pulled main and modify relevant files, causing `yarn generate-api-types` to run since we are running the pre-commit hooks.

**Comment by bttf** [comment]
@tinahollygb Thanks - I don't think the problem stemmed from people avoiding the pre-commit hooks but rather our pre-commit hooks not catching model changes, which would lead to these out-of-sync states. This issue wouldn't reveal itself until someone else later ran the generate command. In any case, adding to the CI flow would add another measure to prevent this from happening so I made those changes and tested.

**Comment by bttf** [comment]
@jdorn @tinahollygb I believe this is ready for re-review ... Thanks!

**Comment by tinahollygb** [comment]
> Thanks - I don't think the problem stemmed from people avoiding the pre-commit hooks but rather our pre-commit hooks not catching model changes, which would lead to these out-of-sync states.

@bttf it's both actually. There's multiple concerns: people not running the hooks, and also the changes your PR initially covers. You'll see on main currently if you make a random JS change and commit it, if you have pre-commit hooks enabled, lint will run and you'll commit some unrelated changes due to the latest changes in one of the components. I see the new changes you made should hopefully catch that, assuming file changes from the lint task will carry through to the task you added.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -62,6 +62,11 @@ jobs:
         env:
           CI: true
 
+      - name: Check api types
+        run: |
+          yarn generate-api-types
+          [[ -n $(git status --porcelain) ]] && echo "You need to run yarn generate-api-types" && exit 1 || true
```
Comment (tinahollygb) [inline .github/workflows/ci.yml:68]: Assuming it'll catch lint tasks above, maybe let's change the message to include `or yarn lint`

**Review by tinahollygb — APPROVED** [review]
Suggested copy change that might lead to developer confusion. Alternatively we can change the lint task that runs on CI to not perform any fixes (that's how I usually do it).

**Comment by bttf** [comment]
@tinahollygb Thanks for the suggestion. I altered our CI workflow to run our lint command without `--fix` as you suggested (cc @jdorn). Will merge if there are no remaining concerns!

**Pull request merged**